capella_model_explorer/__main__.py

```diff
@@ -209,124 +205,160 @@ def build_image():
+    logger.info("Installing pre-commit hooks...")
     subprocess.check_call([pre_commit, "install-hooks"])
     subprocess.check_call([pre_commit, "install"])
 
 
 def _find_exe(name: str) -> str:
-    path = shutil.which(name)
-    if path is None:
+    if not (path := shutil.which(name)):
         raise SystemExit(f"Cannot find {name!r}, install it and try again")
     return path
 
 
-def main():
-    parser = argparse.ArgumentParser(
-        description=(
-            "CLI for managing the application. Every command"
-            " has its own `--help` option."
+@click.group()
+def main() -> None:
+    """CLI for managing the Capella Model Explorer application."""
+
+
+@main.command()
+@click.option(
+    "--container",
+    is_flag=True,
+    help="Launch as a Docker container.",
+)
+@click.option(
+    "--dev",
+    is_flag=True,
+    help="Launch in development mode with full auto-reload.",
+)
+@click.option(
+    "--host",
+    envvar="CME_HOST",
+    default=c.Defaults.host,
+    help=f"The hostname or IP address to bind to. Default: {c.Defaults.host}. "
+    "Ignored when running with '--container'.",
+)
+@click.option(
+    "--port",
+    envvar="CME_PORT",
+    default=c.Defaults.port,
+    help=f"The port to listen on. Default: {c.Defaults.port}.",
+)
+@click.option(
+    "--model",
+    envvar="CME_MODEL",
+    default=c.Defaults.model,
+    help="The Capella model to load (file, URL or JSON string). "
+    f"Default: {c.Defaults.model}.",
+)
+@click.option(
+    "--templates-dir",
+    envvar="CME_TEMPLATES_DIR",
+    default=str(c.Defaults.templates_dir.resolve()),
+    help="The directory containing the templates. "
+    f"Default: {c.Defaults.templates_dir.resolve()}.",
+)
+@click.option(
+    "--live-mode",
+    envvar="CME_LIVE_MODE",
+    default="1" if c.Defaults.live_mode else "0",
+    type=click.Choice(["0", "1"]),
+    help="Control automatic reloading of templates on changes. "
+    "Set to 1 to enable, 0 to disable. "
+    f"Default: {'1' if c.Defaults.live_mode else '0'}. "
+    "Ignored in '--dev' mode.",
+)
+@click.option(
+    "--route-prefix",
+    envvar="CME_ROUTE_PREFIX",
+    default="",
+    help="Add a prefix to all web routes. "
+    "(Note: this prefix does not apply to '/metrics').",
+)
+@click.option(
+    "--docker-image-name",
+    envvar="CME_DOCKER_IMAGE_NAME",
+    default=c.Defaults.docker_image_name,
+    help="The Docker image to use with '--container'. "
+    f"Default: {c.Defaults.docker_image_name}.",
+)
+def run(
+    container: bool,  # noqa: FBT001
+    dev: bool,  # noqa: FBT001
+    host: str,
+    port: int,
+    model: str,
+    templates_dir: str,
+    live_mode: str,
+    route_prefix: str,
+    docker_image_name: str,
+) -> None:
+    """Run the application."""
+    os.environ["CME_HOST"] = host
+    os.environ["CME_PORT"] = str(port)
+    os.environ["CME_MODEL"] = model
+    os.environ["CME_TEMPLATES_DIR"] = templates_dir
+    os.environ["CME_LIVE_MODE"] = live_mode
```

Comment:
The value for `CME_LIVE_MODE` is a string on master because env vars are always strings. The values for boolean env var switches are often set via `"0"` or `"1"`.

Using `click` we can work with a real boolean flag (compare with `dev` and `container`).

Don't forget to adapt the `click.option` accordingly.
```suggestion
    live_mode: bool,
    route_prefix: str,
    docker_image_name: str,
) -> None:
    """Run the application."""
    os.environ["CME_HOST"] = host
    os.environ["CME_PORT"] = str(port)
    os.environ["CME_MODEL"] = model
    os.environ["CME_TEMPLATES_DIR"] = templates_dir
    os.environ["CME_LIVE_MODE"] = "1" if live_mode else "0"
```

Reply:
This seems to work:

```text
M capella_model_explorer/__main__.py
@@ -268,13 +268,11 @@ def main() -> None:
     f"Default: {c.Defaults.templates_dir.resolve()}.",
 )
 @click.option(
-    "--live-mode",
+    "--live-mode/--no-live-mode",
     envvar="CME_LIVE_MODE",
-    default="1" if c.Defaults.live_mode else "0",
-    type=click.Choice(["0", "1"]),
-    help="Control automatic reloading of templates on changes. "
-    "Set to 1 to enable, 0 to disable. "
-    f"Default: {'1' if c.Defaults.live_mode else '0'}. "
+    default=True,
+    help="Control automatic reloading of templates on changes."
+    f" Default: {c.Defaults.live_mode}. "
     "Ignored in '--dev' mode.",
 )
 @click.option(
@@ -298,7 +296,7 @@ def run(
     port: int,
     model: str,
     templates_dir: str,
-    live_mode: str,
+    live_mode: bool,
     route_prefix: str,
     docker_image_name: str,
 ) -> None:
@@ -307,7 +305,7 @@ def run(
     os.environ["CME_PORT"] = str(port)
     os.environ["CME_MODEL"] = model
     os.environ["CME_TEMPLATES_DIR"] = templates_dir
-    os.environ["CME_LIVE_MODE"] = live_mode
+    os.environ["CME_LIVE_MODE"] = "1" if live_mode else "0"
     os.environ["CME_ROUTE_PREFIX"] = route_prefix
     os.environ["CME_DOCKER_IMAGE_NAME"] = docker_image_name
     importlib.reload(c)
```

Reply:
I took the liberty to update your patch a bit and implement some of my opinions as well - I changed the help text slightly, and we're pulling the default from `c.Defaults` again.